### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by yutongz**
Documenting https://github.com/istio/test-infra/pull/662

### CODE REVIEW

File: tests/e2e/README.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ E2E-suites can be manually triggered in PR stage with specific command:
 
 `/test e2e-cluster_wide-auth`
 
+> NOTE: You have to wait "prow/istio-presubmit" to succeed before you can trigger e2e tests against specific commit/SHA. It needs artifacts generated by istio-presubmit.
```
Comment (hklai) [inline tests/e2e/README.md]: Can /test track this dependency automatically?

If not, I'd suggest the following edit.

NOTE: Please do "/test e2e-*" only after "prow/istio-presubmit" completes, as the e2e tests need the artifacts generated by istio-presubmit.

**Comment by mandarjog** [comment]
This is awesome @yutongz @hklai !
Thanks a lot.

**Comment by yutongz** [comment]
OK, it works. Not automatically triggering. Can be kicked off manually.

File: tests/e2e/README.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 This directory contains Istio end-to-end tests and associated test framework.
 
+The whole package of E2E tests are not automatically triggered in Before-Merge stage for every PR except a smoke-test (bookinfo test with auth enabled). After-Merge stage is the place to run all E2E tests.
```
Comment (hklai) [inline tests/e2e/README.md]: Suggestion:

Only an E2E smoke test (bookinfo test with auth enabled) is automatically triggered in Before-Merge stage for every PR. The full suite of E2E tests are run in After-Merge stage only. Their results can be found in https://prow.istio.io/ and https://k8s-testgrid.appspot.com/istio

File: tests/e2e/README.md [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+
+E2E-suites can be manually triggered in PR stage with specific command:
+
+`/test e2e-suite-rbac-no_auth`
+
+`/test e2e-suite-rbac-auth`
+
+`/test e2e-cluster_wide-auth`
+
+> NOTE: Please do "/test e2e-*" only after "prow/istio-presubmit" completes, as the e2e tests need the artifacts generated by istio-presubmit.
```
Comment (hklai) [inline tests/e2e/README.md]: So this can be be automatic (i.e. /test waits for presubmit to be done before proceed)?

Comment (yutongz) [reply tests/e2e/README.md]: No, it won't. That's why I use your suggestion before.

**Pull request merged**